'getchar' is a C reserved identifier, but not necessarily an external symbol
(it's a macro on NetBSD-6.0).

--- src/tools/io/vpKeyboard.h~	2011-09-27 10:03:14.000000000 +0200
+++ src/tools/io/vpKeyboard.h	2012-11-05 10:43:45.370274211 +0100
@@ -94,7 +94,7 @@
   vpKeyboard();
   ~vpKeyboard();
   int kbhit();
-  int getchar();
+  int vp_getchar();
 
 private:
   void init();
--- src/tools/io/vpKeyboard.cpp~	2012-11-05 10:40:17.015276545 +0100
+++ src/tools/io/vpKeyboard.cpp	2012-11-05 11:07:50.014784788 +0100
@@ -72,10 +72,10 @@
   Get the hit key. kbhit() indicates if a key was hitten.
 */
 int
-vpKeyboard::getchar()
+vpKeyboard::vp_getchar()
 {
   int c;
-  c = ::getchar();
+  c = getchar();
   return c;
 }
 
--- example/tools/keyboard.cpp~	2011-02-11 14:17:26.000000000 +0100
+++ example/tools/keyboard.cpp	2012-11-05 10:44:47.877974210 +0100
@@ -69,7 +69,7 @@
     while (1) {
 
       if (keyboard.kbhit()) {
-	c = keyboard.getchar () ;
+	c = keyboard.vp_getchar () ;
 	printf("You hit key: %d '%c'\n", c, c);
 	if (c == 'q' || c == 'Q') {
 	  printf("You hit key: %d %c we stop the loop\n", c, c);
